### DISCUSSION TITLE
Select (new field) allow null/empty input?

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#132] [Q&A] [answered]
I am testing the new `Select` field, love it so far!
Just wondering what the best practice is for allowing a user to clear that field...

- I could add an empty value option to the `data` array...
- I could add an `actionIcon` via `rightSection` with a function to just clear the value?
- or the element could allow a user to clear the selection?

Even with `searchable` enabled, unless the user-entered content matches one of the options - it will auto-select the last chosen option. I feel like if the user clears the text entirely, it should revert back to the uninitialised state (with placeholder).

**Comment by rtivital** [comment, marked as answer]
That's a good point, right section cannot be used since it is occupied by chevron icon, so i guess the best strategy will be to:
- Add prop `clearable`
- Based on that prop render button to clear field in the right section instead of chevron when value is selected
- When select searchable it would also clear the value when input contents is deleted

What do you think?

**Reply by cstrat** [reply]
That sounds like a perfect solution, use the `<CloseButton />` action icon. Very good!
